lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -34,6 +41,67 @@
 )
 
 
+@dataclass
+class Type:
+    """Link query filter type data
+
+    Link the filter type name, the equivalent SQLAlchemy type, and the Pbench
+    API conversion method to create a compatible object from a string.
+
+    Fields:
+        sqtype: The base SQLAlchemy type corresponding to a Python target type
+        convert: The Pbench API conversion method to verify and convert a
+                string
+    """
+
+    sqltype: Any
+    convert: Callable[[str, Any], Any]
+
+
+"""Associate the name of a filter type to the Type record describing it."""
+TYPES = {
+    "bool": Type(Boolean, convert_boolean),
+    "date": Type(TZDateTime, convert_date),
+    "int": Type(Integer, convert_int),
+    "str": Type(String, convert_string),
+}
+
+
+"""Define the set of operators we allow in query filters.
+
+This maps a symbolic name to a SQLAlchemy filter method on the column.
+"""
+OPERATORS = {
+    "~": "contains",
+    "=": "__eq__",
+    "<": "__lt__",
+    ">": "__gt__",
+    "<=": "__le__",
+    ">=": "__ge__",
+    "!=": "__ne__",
+}
+
+
+def make_operator(
+    expression: ColumnElement, operator: str, value: Any
+) -> BinaryExpression:
+    """Return the SQLAlchemy operator method of a column or JSON expression.
+
+    Args:
+        expression: A SQLAlchemy expression or column
+        operator: The operator prefix
+        value: The value to be compared against
+
+    Returns:
+        A SQLAlchemy filter expression.
+    """
+    x = getattr(expression, OPERATORS[operator])(value)
+    current_app.logger.info(
+        "Expression type {} yields type {}", type(expression).__name__, type(x).__name__
+    )
+    return x
```

Comment:
Debugging code, trying to find what types to use; this should be just `return getattr(...)`